nodeJS/express/express_forms_validation.md

```diff
@@ -0,0 +1,565 @@
+### Introduction
+
+In this lesson, we'll dive into forms and how to manage them using Express. We'll explore ways to secure your Node.js / Express applications by using `express-validator` for validation and sanitization. Understanding proper form handling is crucial for maintaining data integrity and protecting your web applications from security risks.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Understanding HTML form elements and their actions.
+- Using `express-validator` for validation and sanitization.
+- Best practices for form design and data integrity.
+
+### HTML forms overview
+
+First, a brief refresher on the actual HTML.
+
+Let's create a simple HTML form, with a single text field for collecting a full name, and its associated label element.
+
+The HTML structure would look something like this:
+
+```html
+<form action="/users/create" method="POST" >
+  <label for="fullName">Full Name:</label>
+  <input placeholder="John Doe" type="text" name="fullName" id="fullName">
+  <button type="submit">Submit</button>
+</form>
+```
+
+We have the HTML form element itself, with an action pointing to some resource on our server, and a method defined. Notice how the method corresponds to an HTTP verb, typically either `GET` or `POST`.
+
+The input needs a corresponding label element, which is what the end user actually sees on the page.
+
+Then, we have our input. It has an `id` and a `name` attribute, which must match the `for` attribute defined on our label above. The `name` attribute is especially important, as it defines how our input will be identified in the form data sent to our server.
+
+It has a type of `text` which tells the browser what type of widget to display, what values to accept, and is used for JavaScript and CSS selectors.
+
+The `type="submit"` button can be pressed by the user to upload the data entered into all other input elements to the server (in this case there's only one).
```

Comment:
I feel like the small details here are unnecessary. Learners will have used forms, labels and inputs already, so we don't need to explain about the purpose of the label, its `for` attribute etc.

We can streamline this bit to just say we have a form with a text input, a label for it, and a submit button. Then we can include the details on the importance of the `name` field, since that's the most important thing we haven't really used in the curriculum yet.